src/pages/Apps/New/:provider/Provider.github.js

```diff
@@ -110,43 +112,89 @@ export default class GithubRepos extends PureComponent {
       width: 1000,
       onClose: async () => {
         const { accounts, selectedAccount } = await this.getInstallations();
-        this.updateState({ accounts, selectedAccount, loading: true });
-        this.getRepositories(selectedAccount);
+        this.updateState({ accounts, selectedAccount });
+        this.getRepositories();
       },
     });
   };
 
-  getRepositories = async (account) => {
-    if (!account) {
+  getRepositories = async () => {
+    // set loading to true to show a spinner until api requests are done
+    this.updateState({ loading: true });
+
+    // get the selected account
+    const { selectedAccount } = this.state;
+
+    if (!selectedAccount) {
       return this.updateState({ loading: false });
     }
 
-    const installationId = account.installationId;
     const api = this.props.github;
-    const res = await api.repositories({ installationId });
+
+    // get current pageIndex and items per page
+    const { repositories, pageQueryParams } = this.state;
+
+    // api params
+    const defaultParams = {
+      page: 1,
+      per_page: 100,
+    };
+    const params =
+      Object.keys(pageQueryParams).length > 0 ? pageQueryParams : defaultParams;
+
+    // fetch the repositories
+    const res = await api.repositories({
+      installationId: selectedAccount.installationId,
+      params,
+    });
+
+    // adds the current page to the repositories
+    const page = res.repositories;
+    repositories.push(...page);
+
+    // check if there is a next page with more repositories
+    const hasNextPage =
+      res.total_count > pageQueryParams.page * pageQueryParams.per_page;
+
+    // gather the params for the next page (if there is one) into an object for consecutive calls through `load more` button
+    const nextPageParams = hasNextPage
+      ? {
+          ...pageQueryParams,
+          page: pageQueryParams.page + 1,
+        }
+      : defaultParams;
 
     this.updateState({
-      selectedAccount: account,
-      repositories: res.repositories,
+      repositories: repositories,
```

Comment:
We can use something like in order not to mutate the repositories state object:

```
repositories: [].concat(...repositories, ...res.repositories);
```

Reply:
this will mutate the state aswell because the const repositories represents the state, doesn't it? what about defining a new variable to hold the repositories?

Reply:
Much better!